### PR TITLE
Remove "logger" field from config.json

### LINKED ISSUE #5176: Remove "logger" field from config.json
Remove "logger" field from config.json. Support for logging to a file needs to be removed as part of the improvements we are making to server side logging. 

The changes that need to be made are
1. Change the config.json structure version from 20 to 21.
2. Upgrade the structure from 20 to 21 upon startup.
3. Make sure that the unit tests reflect this change.
4. Add structure version 21.
5. Remove file-logger.go

### PR DESCRIPTION
**Pull request by kannappanr**
File logging is removed as part of improvement to server logging.

Change the config.json structure version from 20 to 21.

Upgrade the config.json structure to version 21 upon startup.

Added unit tests to cover this change.

Fixes #5176

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/config-current.go [hunk 1]
```diff
@@ -71,6 +71,23 @@ func (s *serverConfig) GetRegion() string {
...
+
+// GetDomain gets current domain.
+func (s *serverConfig) GetDomain() string {
+	s.RLock()
+	defer s.RUnlock()
+
+	return s.Domain
+}
+
 // SetCredentials set new credentials. SetCredential returns the previous credential.
```
Comment (ebozduman) [inline cmd/config-current.go]: `// SetCredentials set new credentials. SetCredential returns the previous credential.`
 =>
`// SetCredential sets the new credentials and returns the previous credential.`

File: cmd/config-versions.go [hunk 2]
```diff
@@ -509,3 +509,19 @@ type serverConfigV20 struct {
 	// Notification queue configuration.
 	Notify *notifier `json:"notify"`
 }
+
+// serverConfigV21 server configuration version '21' which is like
```
Comment (ebozduman) [inline cmd/config-versions.go]:
```
// serverConfigV21 server configuration version '21' which is like
// version '20' except it removes logger field
```
 => 
```
// serverConfigV21 is just like version '20' without logger field
```

**Review by ebozduman — CHANGES_REQUESTED** [review]
Just commented about 2 comments.

File: cmd/config-current.go (hunk 1 above)
Comment (kannappanr) [reply cmd/config-current.go]: done

File: cmd/config-versions.go (hunk 2 above)
Comment (kannappanr) [reply cmd/config-versions.go]: done

**Comment by kannappanr** [comment]
@krishnasrinivas @ebozduman Can you take a look?

**Review by krishnasrinivas — DISMISSED** [review]

**Review by ebozduman — DISMISSED** [review]

**Comment by deekoder** [comment]
codecoverage needs to complete.

**Comment by kannappanr** [comment]
Code coverage is saying ` Waiting for status to be reported ` . Will close this PR and open a new one to see if this status goes away

**Pull request closed without merging**